src/lib/mina_lib/mina_lib.ml

```diff
@@ -2193,6 +2283,28 @@ let create ~commit_id ?wallets (config : Config.t) =
           in
           (* tie other knot *)
           sync_status_ref := Some sync_status ;
+          O1trace.background_thread "fetch_completed_snarks" (fun () ->
+              let open Deferred.Let_syntax in
+              let rec loop () =
+                let status = !sync_status_ref in
+                if Option.is_none status then loop ()
+                else
+                  match
+                    Mina_incremental.Status.Observer.value
+                      (Option.value_exn status)
+                  with
+                  | Ok `Offline | Ok `Bootstrap ->
+                      let%bind () = after (Time.Span.of_sec 1.) in
+                      loop ()
+                  | Ok `Synced ->
```

Comment:
A couple of issues using sync status-
1. Node goes to sync status 4 blocks before it fully syncs- https://github.com/MinaProtocol/mina/blob/d84b8163094e8f92e3021b99d6800262c5b53ce4/src/lib/mina_commands/mina_commands.ml#L394. I believe this was added to avoid being in catchup state for short catchups or say when it hears a new block but it is still being processed. 
2. A node could get into sync status for a brief period during initialization just before it hears of a new block. Bootstrap or catchup is triggered next.

How about waiting for a specific change in status? Like, from `Connecting` to `Bootstrap`or `Catchup` and then synced at  `max-observed-block-height` -1?

Reply:
Since we fetch best tip during bootstrap, can't we assume the max block height will always be a value that is relatively acceptable wrt to the actual tip of the chain? I feel accepting some lag is the simplest route, the worst case is we fail to verify some snarks because the relevant account information at `max_block - 5` isn't in the ledger yet. I'm willing to accept this, given that this is a best effort solution anyway. My assumption is that _most_ of the completed snark work we fetch should still be verifiable given that they are for accounts that are already created.

Reply:
Snarks are also checked against statements in the scan state and these won't be available if blocks aren't processed and added to the transition frontier. So any snark referencing statements in the latest set of blocks will be rejected i.e., snarks for new work added in the latest 4-5 blocks could be rejected. These are the ones BPs will need to include coinbase and user transactions
This would have been ok if `Synced` status was set at, say `max-block-height = max-observed-block-height -1` as opposed to 5. BPs producing blocks would have necessary snarks for immediate slots as work-delay is set to two blocks. 

Alternatively, a node could do a periodic sync (one soon after the initial sync)